src/core/Jalasoft.TeamUp.Resumes.Core/ResumesService.cs

```diff
@@ -17,17 +18,48 @@ public ResumesService(IRepository<Resume> resumesRepository)
 
         public Resume GetResume(Guid id)
         {
-            return this.resumesRepository.GetById(id);
+            try
+            {
+                var resume = this.resumesRepository.GetById(id);
+                if (resume == null)
+                {
+                    throw new ResumeException(404);
+                }
+
+                return resume;
+            }
+            catch (ResumeException ex)
+            {
+                throw ex;
+            }
+            catch (Exception)
+            {
+                throw new ResumeException(500);
+            }
         }
 
         public Resume[] GetResumes()
         {
-            return this.resumesRepository.GetAll().ToArray();
+            try
+            {
+                return this.resumesRepository.GetAll().ToArray();
+            }
+            catch (Exception)
+            {
+                throw new ResumeException(500);
```

Comment:
I know this is not necessary for this US but, maybe using enum instead of 500 number could be a great addition.